api/src/domain/user/updateOperatingPhase.test.ts

```diff
@@ -84,20 +118,40 @@ describe("updateOperatingPhase", () => {
           operatingPhase: "NEEDS_TO_FORM",
           legalStructureId: "general-partnership",
         }),
-        taskProgress: { [formationTaskId]: "IN_PROGRESS" },
+        taskProgress: { [formationTaskId]: "IN_PROGRESS", [businessStructureTaskId]: "COMPLETED" },
       });
       expect(updateOperatingPhase(userData).profileData.operatingPhase).toBe("NEEDS_TO_REGISTER_FOR_TAXES");
     });
+
+    it("updates status back to NEEDS_BUSINESS_STRUCTURE if business structure task is not completed", () => {
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessPersona: "STARTING",
+          operatingPhase: "NEEDS_TO_FORM",
+          legalStructureId: "limited-liability-company",
+        }),
+        taskProgress: {
+          [businessStructureTaskId]: "IN_PROGRESS",
+          [formationTaskId]: "COMPLETED",
+          [taxTaskId]: "COMPLETED",
+        },
+      });
+      expect(updateOperatingPhase(userData).profileData.operatingPhase).toBe("NEEDS_BUSINESS_STRUCTURE");
+    });
   });
 
   describe("NEEDS_TO_REGISTER_FOR_TAXES", () => {
-    it("updates status to FORMED_AND_REGISTERED when tax task and formation task is completed", () => {
+    it("updates status to FORMED_AND_REGISTERED when tax, formation, and business structure task are completed", () => {
```

Comment:
i wonder if we could simplify these tests to be more declarative now that we did that amazing refactor to simplify the code to be more declarative

Reply:
as a follow up refactor, doesn't need to happen before merging code